packages/core/src/icon/icon.element.ts

```diff
@@ -167,6 +175,13 @@ export class CdsIcon extends LitElement {
     if (props.has('title')) {
       this.updateSVGAriaLabel();
     }
+
+    if (props.has('innerOffset') && this.innerOffset > 0) {
+      const dimension = `calc(100% + ${this.innerOffset * 2}px)`;
+      this.svg.style.width = dimension;
+      this.svg.style.height = dimension;
+      this.svg.style.margin = `-${this.innerOffset} 0 0 -${this.innerOffset}`;
+    }
```

Comment:
Some of the icons have whitespace around the svg itself which causes excess whitespace between elements. This lets us define a whitespace offset so if we define a icon to be 16px the icon is edge to edge of the 16px host cds-icon element with no extra white space.

Maybe this is something we could fix in the future with the SVGs as a breaking change? Or make it more automated to opt into whitespace removal? For now I kept this as a internal API.

Reply:
The SVGs all have some white space around the edge. This is because curved paths at the very edge of the Vbox will be cut off. I don't think we can get rid of that.

The white space is consistent, however. It's a ring of 1px White space at the full 36 x 36 px. So that ratio would scale up or down depending on the size of the icon.

I think this is fine how it is being done here. I like it better than the CSS hack. But it is absolutely possible for us to guesstimate a default offset. `needFnToRoundToNearestHundredth(0.27778 * intendedSizeOfIcon)`;